cluster-operator/src/main/java/io/strimzi/operator/cluster/crd/model/AbstractSsLike.java

```diff
@@ -0,0 +1,131 @@
+package io.strimzi.operator.cluster.crd.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.fabric8.kubernetes.api.model.Affinity;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.KubeLink;
+import io.strimzi.crdgenerator.annotations.Minimum;
+
+import java.util.Map;
+
+/**
+ * A baseclass for {@link Kafka} and {@link Zookeeper} with all their common properties.
+ */
+public abstract class AbstractSsLike {
+
+    protected int replicas;
+
+    protected String image;
+
+    protected Resources resources;
+
+    protected Probe livenessProbe;
+
+    protected Probe readinessProbe;
+
+    protected JvmOptions jvmOptions;
+
+    protected Map<String, Object> metrics;
+
+    protected Storage storage;
+
+    protected Affinity affinity;
+
+    @Description("The number of pods in the cluster.")
+    @Minimum(1)
+    @JsonProperty(required = true)
+    public int getReplicas() {
+        return replicas;
+    }
+
+    public void setReplicas(int replicas) {
+        this.replicas = replicas;
+    }
+
+    @Description("The docker image for the pods.")
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    public String getImage() {
+        return image;
+    }
+
+    public void setImage(String image) {
+        this.image = image;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Description("Resource constraints (limits and requests).")
+    public Resources getResources() {
+        return resources;
+    }
+
+    public void setResources(Resources resources) {
+        this.resources = resources;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @Description("Pod liveness checking.")
+    public Probe getLivenessProbe() {
+        return livenessProbe;
+    }
+
+    public void setLivenessProbe(Probe livenessProbe) {
+        this.livenessProbe = livenessProbe;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    @Description("Pod readiness checking.")
+    public Probe getReadinessProbe() {
+        return readinessProbe;
+    }
+
+    public void setReadinessProbe(Probe readinessProbe) {
+        this.readinessProbe = readinessProbe;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @Description("JVM Options for pods")
+    public JvmOptions getJvmOptions() {
+        return jvmOptions;
+    }
+
+    public void setJvmOptions(JvmOptions jvmOptions) {
+        this.jvmOptions = jvmOptions;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Description("The Prometheus JMX Exporter configuration. " +
+            "See https://github.com/prometheus/jmx_exporter for details of the structure of this configuration.")
+    public Map<String, Object> getMetrics() {
+        return metrics;
+    }
+
+    public void setMetrics(Map<String, Object> metrics) {
+        this.metrics = metrics;
+    }
+
+    @Description("Persistent storage (disk). Cannot be updated.")
```

Comment:
I meant persistent in the sense of it not being RAM, basically. "Storage configuration (disk)." is a fairly useless description though: It's _obviously_ storage configuration, because the property is called "storage" and it's in an object which is about configuring a Kafka or Zookeeper cluster. 

Although `Storage` is a good name for the class perhaps the name of the property should be more along the line `logStorage`, since that's what we put there?

Reply:
There is a lot of people who do not understand that the logs are also the records and not only the Log4j logs. So I think that would be even more confusing. To me it seems that _storage_ is pretty clear - just the _Persistent_ is not 100% correct.